components/functional/CustomScroll.tsx

```diff
@@ -0,0 +1,32 @@
+import { useEffect } from 'react';
+
+// open all external links in another tab
+export default function CustomScroll() {
```

Comment:
```suggestion
export default function useHeaderScroll() {
```
Here what im trying to say about creating a custom hook. Something like that.
What do you think?
```
export default function useHeaderScroll() {
  // logic to get header height
  // your handleLinkClick function
  return { handleLinkClick }
}
```